src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1217,6 +1239,37 @@ private void OutputExternCSignatures()
             Out.Write(sb.ToString());
         }
 
+        /// <summary>
+        /// Output C++ code for a given unboxingStubNode
+        /// </summary>
+        /// <param name="unboxingStubNode">The unboxing node to be output</param>
+        /// <param name="methodImplementations">The buffer in which to write out the C++ code</param>
+        private void OutputUnboxNode(UnboxingStubNode unboxingStubNode)
```

Comment:
The unboxing stubs define their own mangled name. We could derive `CppMethodCodeNode`, providing the unboxing names. Would have to swap `GetCppMethodName` for `method.GetMangledName` in several places I think.